pkg/operator/ceph/cluster/watcher.go

```diff
@@ -201,47 +201,79 @@ func (c *clientCluster) fenceNode(ctx context.Context, node *corev1.Node, cluste
 	}
 	logger.Debugf("volumesInuse %s", volumesInuse)
 
-	rbdVolumesInUse := getCephVolumesInUse(cluster, volumesInuse)
-	if len(rbdVolumesInUse) == 0 {
-		logger.Debugf("no rbd volumes in use for out of service node %q", node.Name)
+	rbdVolumesInUse, cephFSVolumeInUse := getCephVolumesInUse(cluster, volumesInuse)
+	if len(rbdVolumesInUse) == 0 && len(cephFSVolumeInUse) == 0 {
+		logger.Debugf("no rbd or cephFS volumes in use for out of service node %q", node.Name)
 		return nil
 	}
 
-	logger.Infof("node %q require fencing, found rbd volumes in use", node.Name)
 	listPVs, err := c.context.Clientset.CoreV1().PersistentVolumes().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return pkgerror.Wrapf(err, "failed to list PV")
 	}
 
-	rbdPVList := listRBDPV(listPVs, cluster, rbdVolumesInUse)
-	if len(rbdPVList) == 0 {
-		logger.Debug("No rbd PVs found on the node")
-		return nil
-	}
+	if len(rbdVolumesInUse) != 0 {
+		logger.Infof("node %q require fencing, found rbd volumes in use", node.Name)
 
-	clusterInfo, _, _, err := opcontroller.LoadClusterInfo(c.context, ctx, cluster.Namespace, &cluster.Spec)
-	if err != nil {
-		return pkgerror.Wrapf(err, "Failed to load cluster info.")
+		rbdPVList := listRBDPV(listPVs, cluster, rbdVolumesInUse)
+		if len(rbdPVList) == 0 {
+			logger.Debug("No rbd PVs found on the node")
+		} else {
+			clusterInfo, _, _, err := opcontroller.LoadClusterInfo(c.context, ctx, cluster.Namespace, &cluster.Spec)
+			if err != nil {
+				return pkgerror.Wrapf(err, "Failed to load cluster info.")
+			}
+
+			for i := range rbdPVList {
+				err = c.fenceRbdImage(ctx, node, cluster, clusterInfo, rbdPVList[i])
+				// We only need to create the network fence for any one of rbd pv.
+				if err == nil {
+					break
+				}
+
+				if i == len(rbdPVList)-1 {
+					return pkgerror.Wrapf(err, "failed to fence rbd volumes")
+				}
+				logger.Errorf("failed to fence rbd volumes %q, trying next rbd volume", rbdPVList[i].Name)
+			}
+		}
 	}
 
-	for i := range rbdPVList {
-		err = c.fenceRbdImage(ctx, node, cluster, clusterInfo, rbdPVList[i])
-		// We only need to create the network fence for any one of rbd pv.
-		if err == nil {
-			break
+	if len(cephFSVolumeInUse) != 0 {
+		logger.Infof("node %q require fencing, found cephFS volumes in use", node.Name)
+
+		cephFSPVList := listCephFSPV(listPVs, cluster, cephFSVolumeInUse)
```

Comment:
This only returns RWO CephFS volumes, right? It might be helpful to put that in the method name to be clear
```suggestion
		cephFSPVList := listRWOCephFSPV(listPVs, cluster, cephFSVolumeInUse)
```